File: backend/app/features/blocked_sites/models/blocked_site.py
from sqlalchemy import Column, Integer, String, DateTime, Boolean
from datetime import datetime, timezone
from app.shared.database import Base


class BlockedSite(Base):
    __tablename__ = "blocked_sites"
    id = Column(Integer, primary_key=True, index=True)
    domain = Column(String, unique=True, index=True)
    reason = Column(String)
    category = Column(String, nullable=True)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    created_by = Column(Integer, nullable=True)
    updated_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))
    updated_by = Column(Integer, nullable=True)
    is_deleted = Column(Boolean, default=False, nullable=False)

File: frontend/src/features/blocked-sites/types/blockedSite.ts
export interface BlockedSite {
  id: number;
  domain: string;
  reason: string;
  category?: string | null;
  created_at?: string;
  updated_at?: string;
  deleted_at?: string | null;
}

export interface BlockedSiteCreate {
  domain: string;
  reason: string;
  category?: string | null;
}

export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

